web/modules/custom/dpl_opening_hours/src/Mapping/OpeningHoursMapper.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\dpl_opening_hours\Mapping;
+
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGET200Response;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGETRequest;
+use DanskernesDigitaleBibliotek\CMS\Api\Model\DplOpeningHoursGETRequestCategory;
+use Drupal\dpl_opening_hours\Model\OpeningHoursInstance;
+use Drupal\node\NodeStorageInterface;
+use Drupal\taxonomy\TermStorageInterface;
+use Safe\DateTime;
+use Safe\DateTimeImmutable;
+
+/**
+ * Mapper between value objects and OpenAPI request/response objects.
+ */
+class OpeningHoursMapper {
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    private NodeStorageInterface $branchStorage,
+    private TermStorageInterface $categoryStorage,
+  ) {}
+
+  /**
+   * Map an OpenAPI request to a value object.
+   */
+  public function fromRequest(DplOpeningHoursGETRequest $request) : OpeningHoursInstance {
+    $branch = $this->branchStorage->load($request->getBranchId());
+    if (!$branch || $branch->bundle() !== "branch") {
+      throw new \InvalidArgumentException("Invalid branch id '{$request->getBranchId()}'");
+    }
+
+    $categoryTitle = $request->getCategory()?->getTitle();
+    if (!$categoryTitle) {
+      throw new \InvalidArgumentException('No category title provided');
+    }
+    $categoryTerms = $this->categoryStorage->loadByProperties([
+      'name' => $categoryTitle,
+      'vid' => 'opening_hours_categories',
+    ]);
+    $categoryTerm = reset($categoryTerms);
+    if (!$categoryTerm) {
```

Comment:
```suggestion
    if (!($categoryTerm instanceof Term)) {
```

Reply:
Fixed in [66cefda](https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/890/commits/66cefdafd2987b2d16aa676919eaf06bc333477c).